src/constructor/constructor.py

```diff
@@ -0,0 +1,259 @@
+
+        Args:
+            parameters: Parameters for optimization
+            optim_idx: Optimizer and scheduler index if specific optimizer/scheduler group is needed to be created.
+            Default is -1 meaning all the optimizers/schedulers are created
+
+        Returns:
+            List of dicts in the PyTorch Lightning accepted format:
+            - optimizer: PyTorch-like optimizer
+            - lr_scheduler: dict
+                scheduler: PyTorch-like scheduler
+                <any PyTorch Lightning parameters. See `their documentation`_>
+
+        Raises:
+            - ValueError: when a requested optimizer/scheduler group with optim_idx isn't present in configuration
+            - ValueError: when parameters type is out of supported types list (see typing)
+
+        .. _their documentation:
+        https://pytorch-lightning.readthedocs.io/en/1.6.2/common/lightning_module.html#configure-optimizers
+        """
+        optims_params = self.__hparams.optimization
+        if 0 <= optim_idx < len(optims_params):
+            optims_params = [optims_params[optim_idx]]
+        elif optim_idx >= len(optims_params):
+            raise ValueError(f'You requested optimization with index {optim_idx} while '
+                             f'there\'re only {len(optims_params)} optimization parameters are specified')
+
+        opt_sched_list = []
+        for optim_params in optims_params:
+            optimizer = self.__create_optimizer(parameters, optim_params.optimizer)
+            opt_sched = {'optimizer': optimizer}
+
+            if 'scheduler' in optim_params:
+                scheduler_dict = self.__create_scheduler(optimizer, optim_params.scheduler)
+                opt_sched['lr_scheduler'] = scheduler_dict
+
+            opt_sched_list.append(opt_sched)
+
+        return opt_sched_list
+
+    @staticmethod
+    def __create_optimizer(parameters: Union[Module, Tensor, Iterable[Union[Module, Tensor]]],
+                           optim_params: DictConfig) -> Optimizer:
+        optimizer_class = OPTIMIZERS.get(optim_params.optimizer.name)
+        parameters = Constructor.__set_weight_decay_for_parameters(parameters)
+        optimizer = optimizer_class(parameters, **optim_params.optimizer.params)
+
+        return optimizer
+
+    @staticmethod
+    def __create_scheduler(optimizer: Optimizer, scheduler_params: DictConfig) -> Dict[str, Any]:
+        scheduler_class = SCHEDULERS.get(scheduler_params.name)
+        scheduler = scheduler_class(optimizer, **scheduler_params.params)
+        pl_params = scheduler.pl_params if 'pl_params' in scheduler else {}
+
+        return {
+            'scheduler': scheduler,
+            **pl_params
+        }
+
+    @staticmethod
+    def __set_weight_decay_for_parameters(parameters: Union[Module, Tensor, Iterable[Union[Module, Tensor]]]
+                                          ) -> List[Dict[str, Any]]:
```

Comment:
In google style it write like
`def __set_weight_decay_for_parameters(parameters: Union[Module, Tensor, Iterable[Union[Module, Tensor]]]
  ) -> List[Dict[str, Any]]`